src/pipecat/transports/services/pipecat_cloud.py

```diff
@@ -0,0 +1,313 @@
+        input_name: Optional name for the input transport.
+        output_name: Optional name for the output transport.
+    """
+
+    # Event name mappings for each transport type
+    # Only include events that need to be mapped differently
+    _EVENT_MAPPINGS = {
+        DailyTransport: {
+            "on_client_connected": "on_first_participant_joined",
+            "on_client_disconnected": "on_participant_left",
+        },
+    }
+
+    def __init__(
+        self,
+        session_args: PipecatCloudSessionArguments,
+        params: Optional[Union[PipecatCloudParams, TransportParams]] = None,
+        *,
+        input_name: Optional[str] = None,
+        output_name: Optional[str] = None,
+    ):
+        super().__init__(input_name=input_name, output_name=output_name)
+
+        # Convert TransportParams to PipecatCloudParams if needed
+        if isinstance(params, TransportParams):
+            cloud_params = PipecatCloudParams()
+            for field_name, field_value in params.model_dump().items():
+                setattr(cloud_params, field_name, field_value)
+            params = cloud_params
+        else:
+            params = params or PipecatCloudParams()
+
+        self._pending_handlers = {}
+
+        # Create the appropriate transport based on session arguments type
+        args = session_args.args
+        if isinstance(args, WebSocketSessionArguments):
+            logger.info("Using FastAPIWebsocketTransport")
+            websocket_params = params.to_websocket_params()
+            self._transport = FastAPIWebsocketTransport(
+                args.websocket,
+                websocket_params,
+                input_name=input_name,
+                output_name=output_name,
+            )
+        elif isinstance(args, DailySessionArguments):
+            logger.info("Using DailyTransport")
+            daily_params = params.to_daily_params()
+            self._transport = DailyTransport(
+                args.room_url,
+                args.token,
+                # TODO-CB: Bot name is missing from DailySessionArguments
+                "Bot",
+                params=daily_params,
+                input_name=input_name,
+                output_name=output_name,
+            )
+        elif isinstance(args, WebRTCSessionArguments):
+            logger.info("Using SmallWebRTCTransport")
+            transport_params = params.to_transport_params()
+            self._transport = SmallWebRTCTransport(
+                args.webrtc_connection,
+                transport_params,
+                input_name=input_name,
+                output_name=output_name,
+            )
+        else:
+            raise ValueError(f"Unsupported session arguments type: {type(args)}")
+
+        # Register any handlers that were added before transport creation
+        for event_name, handlers in self._pending_handlers.items():
+            for handler in handlers:
+                self._register_handler(event_name, handler)
```

Comment:
I don't think this is needed. The transport is created in the constructor, so it's impossible to have pending handlers because the constructor hasn't finished yet.